### PR TITLE
change state if RITM state changes

### PR DESCRIPTION
**Pull request by Adil404**
This business rule is to set state of sc_task table when sc_req_item state changes.
Create before insert/update business rule on sc_req_item table and give condition if state changes to 2 OR cat_item.sc.catalogs is "take your choice".

### CODE REVIEW

File: Business Rules/RITM state change/Related_task_state_update.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+//create business rule on sc_req_item table and add condtion 
+
+(function executeRule(current, previous /*null when async*/ ) {
+
+    var gr = new GlideRecord("sc_task");
+    gr.addQuery('request_item', current.sys_id);
+    gr.orderBy('order');
+    gr.setLimit(1);
+    gr.query();
+    if (gr.next() && gr.getValue('state') == 1) {
```
Comment (msdfib) [inline Business Rules/RITM state change/Related_task_state_update.js:10]: If you added this condition to the query, you would not need to check the state for '1' (the magic number).

File: Business Rules/RITM state change/Related_task_state_update.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+//create business rule on sc_req_item table and add condtion 
+
+(function executeRule(current, previous /*null when async*/ ) {
+
+    var gr = new GlideRecord("sc_task");
+    gr.addQuery('request_item', current.sys_id);
+    gr.orderBy('order');
+    gr.setLimit(1);
```
Comment (msdfib) [inline Business Rules/RITM state change/Related_task_state_update.js:8]: What if more than one sc_task is under RITM?

File: Business Rules/RITM state change/Related_task_state_update.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+//create business rule on sc_req_item table and add condtion 
+
+(function executeRule(current, previous /*null when async*/ ) {
+
+    var gr = new GlideRecord("sc_task");
+    gr.addQuery('request_item', current.sys_id);
+    gr.orderBy('order');
```
Comment (msdfib) [inline Business Rules/RITM state change/Related_task_state_update.js:7]: You don't need to sort the results if you don't have a clear reason, in this case, you are sorting a list with one record.

**Review by Lacah — CHANGES_REQUESTED** [review]
Have a look at the added comments and resubmit your script accordingly. Thanks.

**Review by Adil404 — COMMENTED** [review]
updated the script : sc_task will change the state from open to work in progress when RITM state changes to work in progress (set this condition in filer condition).

File: Business Rules/RITM state change/Related_task_state_update.js (hunk 3 above)
Comment (Adil404) [reply Business Rules/RITM state change/Related_task_state_update.js]: removed the orderBy query

File: Business Rules/RITM state change/Related_task_state_update.js (hunk 2 above)
Comment (Adil404) [reply Business Rules/RITM state change/Related_task_state_update.js]: removed setLimit(1) so that it will update all the sc_task related to RITM

**Review by Adil404 — COMMENTED** [review]
update the script, please review it and let me know

**Review by msdfib — APPROVED** [review]
LGTM, leaving merge up to @Lacah

**Pull request merged**